cpp/CMakeLists.txt

```
@@ -980,6 +980,9 @@ endif()
 
 if(ARROW_JNI)
   add_subdirectory(src/jni)
+  if(ARROW_DATASET)
+  	add_subdirectory(../java/dataset/src/main/cpp ./java/jni)
+	endif()
```

Please use spaces, not tabs, I think that's why the indentation is off:
```suggestion
  if(ARROW_DATASET)
    add_subdirectory(../java/dataset/src/main/cpp ./java/jni)
  endif()
```